File: frontendapp/lib/models/bookingdetail.dart
class BookingDetail {
  final int? id;
  final int bookingId;
  final int seatId;
  final double price;

  BookingDetail({
    this.id,
    required this.bookingId,
    required this.seatId,
    required this.price,
  });

  factory BookingDetail.fromJson(Map<String, dynamic> json) => BookingDetail(
    id: json['id'],
    bookingId: json['bookingId'],
    seatId: json['seatId'],
    price: json['price'].toDouble(),
  );
}

File: backend/bookingmovieticketapp/app/Models/BookingDetail.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class BookingDetail extends Model
{
    use HasFactory;
    protected $table = 'bookingdetails';
    protected $primaryKey = 'id';
    public $timestamps = false;
    protected $fillable = [
        'bookingid',
        'seatid',
        'price',
    ];
    protected $casts = [
        'price' => 'integer',
    ];

    
    public function booking()
    {
        return $this->belongsTo(Booking::class, 'bookingid');
    }

    public function seat()
    {
        return $this->belongsTo(Seat::class, 'seatid');
    }
}
